app/controllers/QuestionController.scala

```diff
@@ -469,4 +469,30 @@ object QuestionController extends ServiceComponent with MongoController {
 			}
 		}
 	}
+
+	def dashboard_flag(flag: String) = UserAwaredAction.async { implicit context =>
+		val questions = MongoDAO.questionsWithFlag(flag).map {
+			_.toList
+		}
+
+		questions.map { st =>
+			Ok(views.html.dashboard_flag(flag, st.map(q => q.as[Question])))
+		}
+	}
+
+	def dashboard() = UserAwaredAction { implicit context =>
+		val qualityFlags = List("C", "R", "QB", "QG")
+		val contentTypes = List("Q", "H", "S")
+
+		val cross = for {
+			q <- qualityFlags
+			c <- contentTypes
+		// TODO! FIXME!
```

Comment:
Got it, we require [`Future.traverse`](http://stackoverflow.com/questions/20108523/combining-scala-futures-and-collections-in-for-comprehensions).